rsconnect/bundle.py

```diff
@@ -725,7 +725,12 @@ def infer_entrypoint(path, mimetype):
         mimetype_filelist[guess_type(file)[0]].append(rel_path)
         if file in default_mimetype_entrypoints[mimetype]:
             return file
-    return mimetype_filelist[mimetype].pop() if len(mimetype_filelist[mimetype]) == 1 else None
+    res = mimetype_filelist[mimetype].pop() if len(mimetype_filelist[mimetype]) == 1 else None
+    if not res:
+        raise RuntimeError(
```

Comment:
We should consider raising an RSConnectException here. `RuntimeError` is typically raised by the Python language runtime.